app/Models/User.php

```diff
@@ -384,18 +384,29 @@ public static function checkWhenUsernameAvailable($username) : Carbon
 
     public function getUsernameAvailableAt() : Carbon
     {
-        if ($this->group_id !== 2 || $this->user_type === 1) {
-            //reserved usernames
-            return Carbon::now()->addYears(10);
-        }
-
         $playCount = array_reduce(array_keys(Beatmap::MODES), function ($result, $mode) {
             return $result + $this->statistics($mode, true)->value('playcount');
         }, 0);
 
+        if ($this->group_id !== 2) {
+            //reserved usernames
+            return Carbon::now()->addYears(10);  //This will always be in the future, which is wanted
+        }
+
+        if ($this->user_type === 1) {
+            //restricted user
+            if ($playCount > 1000) {
```

Comment:
I think this was a typo, but this conditional probably isn't necessary at all because of the changes made.

Reply:
With this I presume you mean we can use one calculation for both normal and restricted user?

Reply:
Oh nvm, I got it. I will remove the play count > 1000 statement